### ISSUE TITLE
Access controll fails on `ApplicationSubmission#show` with a nonexistant user

### ISSUE THREAD

**Issue by werebus** [umts/jobapps#312]
`@current_user` will be `nil` if the user isn't in our database:

https://github.com/umts/jobapps/blob/4a1db4fbbc64bcdcc8f79f8602b9b1297484ab18/app/controllers/application_controller.rb#L48-L55

[This line][1] should also include a check for `@current_user.present?`

[1]:  https://github.com/umts/jobapps/blob/4a1db4fbbc64bcdcc8f79f8602b9b1297484ab18/app/controllers/application_submissions_controller.rb#L91-L92

**Comment by dfaulken** [comment]
I would write it as:

```ruby
unless @current_user == @record.user || @current_user.try(:staff?)
  deny_access and return
end
```

**Comment by dfaulken** [comment]
I'd also recommend TDD as the way of verifying the bug's fixed.

**Comment by werebus** [comment]
Closed by #341
